demos/minimal/actors/sample_generator.py

```diff
@@ -1,68 +1,79 @@
 from improv.actor import Actor, RunManager
-from datetime import date  # used for saving
 import numpy as np
 import logging
+import time  # Importing time module for the delay
 
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
 
 
 class Generator(Actor):
-    """Sample actor to generate data to pass into a sample processor.
-
+    """Sample actor generate sine/cosine waves based on odd/even frame numbers respectively to pass into a sample processor.
     Intended for use along with sample_processor.py.
     """
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.data = None
         self.name = "Generator"
-        self.frame_num = 0
+        self.frame_num = 0  # Initialize frame counter
+        self.data = None
+        self.max_frames = 20  # Set the limit for number of frames
 
     def __str__(self):
         return f"Name: {self.name}, Data: {self.data}"
 
     def setup(self):
-        """Generates an array that serves as an initial source of data.
-
-        Initial array is a 100 row, 5 column numpy matrix that contains
-        integers from 1-99, inclusive.
-        """
-
+        """Initial setup for Generator"""
         logger.info("Beginning setup for Generator")
-        self.data = np.asmatrix(np.random.randint(100, size=(100, 5)))
+        xs = np.linspace(-10, 10, 100)
+        ys = np.sin(xs)
+        self.data = np.column_stack((xs, ys))
         logger.info("Completed setup for Generator")
 
     def stop(self):
-        """Save current randint vector to a file."""
-
+        """Save current wave vector to file."""
         logger.info("Generator stopping")
         np.save("sample_generator_data.npy", self.data)
         return 0
 
     def runStep(self):
         """Generates additional data after initial setup data is exhausted.
+        
+        Data is a sine wave if the frame number is even or a cosine wave if the frame number is odd."""
+        time.sleep(0.5) # Add a slight pause between frame generation
+        """Sends a flattened array with x and y coordinates followed by frame number."""
```

Comment:
If this is a line comment, you would just have it as `# whatever the comment is` 

The doc string comments using `"""` are only used write after a function or class declaration